### ISSUE TITLE
[Spell][Rogue] Vanish, Stealth

### ISSUE THREAD

**Issue by stealthvanish** [Atlantiss/NetherwingBugtracker#3086]
[//]: # ([Spell][Rogue] Vanish, Stealth)

**Description**: 

Rogue gets out from Vanish/Stealth for reasons he shouldn't.

**Current behaviour**:

1. Vanish/Stealth sometimes fades away from Rogue for another reason than getting damaged/affected by any spell or effect, or removing Vanish/Stealth by himself. https://imgur.com/57R6kuH

2. If someone in anyway executed any kind of damage (for example: autoattack) toward Rogue, but that damage was still not applied on Rogue/didn't affected him; if Rogue uses after that Vanish/Stealth before that damage is even applied to him, the Rogue will lose soon his Vanish/Stealth. For example, in Rogue vs Rogue duel, one Rogue performs Vanish, but then he instantly loses Vanish, as he was affected by a melee autoattack swing that was executed just before the Vanish was even being used. In the process, Rogue can lose important cooldown for nothing, by something he has zero control over.

3. If Rogue uses a keybind, that is supposed to execute a stealth bar ability, too quick, just after using Vanish/Stealth, he will trigger an ability from a non-stealth bar that has the same keybind, thus he will get out of Vanish/Stealth and not utilize this ability in a desired way. For example, Rogue have binded Hemorrhage on "2" for non-stealth bar and the same keybind applies for a Garrote in a stealth bar. Rogue wants to quickly perform Vanish and use Garrote, before getting out of Vanish, as he is dotted by 5 different effects and there is simply no other way to do that without using, for example, CoS to remove magical dots. But because there is some kind of an UI lag (I don't even use any addon for ability bars to avoid this exact issue as much as possible), Rogue can get himself out of Vanish/Stealth by using ability he technically shouldn't be able to use, in this example, he attempted to use Garrote after using Vanish, but he still used somehow Hemorrhage and removed himself out of Vanish.

**Expected behaviour**: 

1.There is a limited amount of factors that can get Rogue out of Vanish/Stealth, if none of them applies to Rogue, he shouldn't lose Vanish/Stealth.

2. Vanish should allow Rogue to avoid any damage that was being executed just before using Vanish, and which still didn't affected Rogue before using it. For example, Death Coil incomes, Rogue uses Vanish, Death Coil doesn't affect him.

3. Rogue shouldn't lose Vanish/Stealth by using abilities he technically shouldn't be able to use. If Rogue performed Vanish/Stealth and his non-stealth bar is supposedly changed to stealth bar, why is he still able to use for some small fraction of time non-stealth bar abilities and get himself out of Vanish/Stealth?

**Server Revision**: 2364

**Comment by Easyfraggy** [comment]
What exactly is that SS supposed to show? There's no timestamp or anything...

**Comment by somnu** [comment]
During TBC era you could have 3 outcomes with vanishing a spell:

- the spell gets absorbed and vanish doesn't break;
- the spell gets absorbed and vanish breaks;
- the spell doesn't get absorbed and vanish breaks.

This all depends on the timing of the incoming spell/vanish.

What you're describing falls either in the 2nd or 3rd category which is blizzlike.

**Comment by stealthvanish** [comment]
> During TBC era you could have 3 outcomes with vanishing a spell:
> 
> * the spell gets absorbed and vanish doesn't break;
> * the spell gets absorbed and vanish breaks;
> * the spell doesn't get absorbed and vanish breaks.
> 
> This all depends on the timing of the incoming spell/vanish.
> 
> What you're describing falls either in the 2nd or 3rd category which is blizzlike.

Go read my initial post once again, because all these "outcomes" can be related to "2nd category" only. The 3rd point is about completely different layer of issue you seem to not understand. It's not about the relation between Vanish and incoming spells, but the fact that lagging UI frames (most likely; without addons) cause Rogue to go out of Vanish by himself.

**Comment by stealthvanish** [comment]
> The screen shot doesn't show anything because without timestamps you can't see what happened - due to spell batching things are not always shown in the order they are executed in the combat log because everything that happens at the same time IE same batch will be shown in a somewhat random order in the combat log.
> 
> From looking at your SS it seems like either Nighteye Panther buff fading breaks your stealth or shadowstep breaks it. But theres no way of knowing without timestamps.
> 
> You can use every ability while stealthed btw - you aren't blocked from using shiv/sinister or any other regular ability when stealthed - there is no "requires no stealth" requirement.

Neither these things break Stealth, because I tried it, but if it would, that would be a serious problem.

I never said you can't use Shiv during Stealth, I said it bring attention to the fact that the combat log shows all outcoming from me and incoming to me effects + I didn't used any opener or anything during Stealth that would get me out of Stealth nor I was affected by anything, yet lost Vanish/Stealth.

**Comment by Evolvee** [comment]
I already said 3) is reported on old bugtracker. No need to waste time discussing it. Its bugged, yes.

**Comment by Easyfraggy** [comment]
So :
1) Need research - unknown cause.
2) Blizzlike
3) Confirmed - reported on old tracker.

**Comment by Zarant** [comment]
>Rogue shouldn't lose Vanish/Stealth by using abilities he technically shouldn't be able to use. If Rogue performed Vanish/Stealth and his non-stealth bar is supposedly changed to stealth bar, why is he still able to use for some small fraction of time non-stealth bar abilities and get himself out of Vanish/Stealth?

This is not a bug, it's a client limitation and it was present in the game back in the day. Your normal bar only changes to the stealth bar once you receive a confirmation from the server that stealth was successfully cast.
Most rogue/druid players back in the day reported the same stealth bar lag, iirc it got fixed much later in MoP.

I found the following comment from September 2008 (patch 2.4):
> When I vanish/stealth in arena(or any other place) to sap my currently blinded target the stealth bar doesn't pop up fast enough so I kick instead(I have kick(when out of stealth) on the same button I have sap(when I'm stealthed))>.< So basically the delay on when the stealth bar overlaps the out-of-stealth bar is way to slow.
https://www.curseforge.com/wow/addons/project-5814

Another post from 2011 (cataclysm) discussing the same issue:

http://www.skill-capped.com/forums/showthread.php?12658-How-to-stop-Rogue-stealth-bar-swapping
Basically players got around that limitation by cleverly setting up their UI to circumvent that issue.

**Comment by Tereneckla** [comment]
Then we're at
 1. Spell Delay (not valid anymore)
 2. Blizzlike
 3. Blizzlike
